packages/react-jss/src/create-use-styles.js

```diff
@@ -0,0 +1,101 @@
+// @flow
+
+import React from 'react'
+import {ThemeContext as DefaultThemeContext} from 'theming'
+
+import {JssContext} from './JssContext'
+import {
+  createStaticSheet,
+  addDynamicRules,
+  removeDynamicRules,
+  updateDynamicRules
+} from './utils/sheets'
+import {getIndex} from './utils/index-counter'
+import type {HookOptions, Styles} from './types'
+import {unmanageSheet, manageSheet} from './utils/managers'
+import {getSheetClasses} from './utils/get-sheet-classes'
+
+const noTheme = {}
+
+function createUseStyles<Theme: {}>(styles: Styles<Theme>, options?: HookOptions<Theme> = {}) {
+  const {index = getIndex(), theming, name = 'Hook', ...sheetOptions} = options
+  const ThemeContext = (theming && theming.context) || DefaultThemeContext
+
+  return (data: any) => {
+    const jssContext = React.useContext(JssContext)
+    // $FlowFixMe
+    const theme: Theme = typeof styles === 'function' ? React.useContext(ThemeContext) : noTheme
+
+    // When the theme or the context changes we create a new sheet
+    const sheet = React.useMemo(
+      () => {
+        if (jssContext.disableStylesGeneration) {
+          return undefined
+        }
+
+        const stylesheet = createStaticSheet({
```

Comment:
we could call it `sheet`

Reply:
ah it's the linter shadow linter issue, I would disable the rule locally here, because it's exactly the same object and it is confusing to have different names for it